docs/src/humann.md

```diff
@@ -1,4 +1,613 @@
+are not associated with a particular taxon.
+
+If you want to hang onto the taxon information,
+use the keyword argument `stratified = true`:
+
+```julia-repl
+julia> gfs_strat = humann_profile("demo_fastq/demo_genefamilies.tsv", stratified=true)
+CommunityProfile{Float64, GeneFunction, MicrobiomeSample} with 1416 features in 1 samples
+
+Feature names:
+UNMAPPED, UniRef90_G1UL42, UniRef90_G1UL42...UniRef90_D0TRR5, UniRef90_D0TRR5
+
+Sample names:
+demo_genefamilies
+
+
+
+julia> first(features(gfs_strat), 5)
+5-element Vector{GeneFunction}:
+ GeneFunction("UNMAPPED", missing)
+ GeneFunction("UniRef90_G1UL42", missing)
+ GeneFunction("UniRef90_G1UL42", Taxon("Bacteroides_dorei", :species))
+ GeneFunction("UniRef90_I9QXW8", missing)
+ GeneFunction("UniRef90_I9QXW8", Taxon("Bacteroides_dorei", :species))
+```
+
+Here, we can see that the uniref90 "G1UL42" was contributed by _Bacteroides dorei_.
+
+The object returned by `humann_profile` is a `CommunityProfile` type from [`Microbiome.jl`](https://github.com/BioJulia/Microbiome.jl),
+and has a bunch of useful properties.
+
+### Indexing
+
+For example, you can index into a `CommunityProfile` just like you would a matrix.
+In julia, you can pull out specific values using `[row, col]`.
+So for example, to get the 3rd row, 2nd column, of matrix `mat`:
+
+```julia-repl
+julia> mat
+4×3 Matrix{Int64}:
+  1   2   3
+  4   5   6
+  7   8   9
+ 10  11  12
+
+julia> mat[3,2]
+8
+```
+
+You can also get "slices", eg to get rows 2-4, column 1:
+
+```julia-repl
+julia> mat[2:4, 1]
+3-element Vector{Int64}:
+  4
+  7
+ 10
+```
+
+To get all of one dimension, you can just use a bare `:`
+
+```julia-repl
+julia> mat[:, 1:2]
+4×2 Matrix{Int64}:
+  1   2
+  4   5
+  7   8
+ 10  11
+```
+
+For `CommunityProfile`s,
+you can index with numbers as above, but also with strings
+representing names of features (rows) or samples (columns):
+
+```julia-repl
+julia> samplenames(gfs_strat)
+1-element Vector{String}:
+ "demo_genefamilies"
+
+julia> featurenames(gfs_strat)
+1416-element Vector{String}:
+ "UNMAPPED"
+ "UniRef90_G1UL42"
+ "UniRef90_G1UL42"
+ "UniRef90_I9QXW8"
+ ⋮
+ "UniRef90_A6LH06"
+ "UniRef90_D0TRR5"
+ "UniRef90_D0TRR5"
+ "UniRef90_D0TRR5"
+
+julia> slice = gfs_strat["UniRef90_D0TRR5", :]
+CommunityProfile{Float64, GeneFunction, MicrobiomeSample} with 3 features in 1 samples
+
+Feature names:
+UniRef90_D0TRR5, UniRef90_D0TRR5, UniRef90_D0TRR5
+
+Sample names:
+demo_genefamilies
+
+
+
+julia> features(slice)
+3-element Vector{GeneFunction}:
+ GeneFunction("UniRef90_D0TRR5", missing)
+ GeneFunction("UniRef90_D0TRR5", Taxon("Bacteroides_vulgatus", :species))
+ GeneFunction("UniRef90_D0TRR5", Taxon("Bacteroides_dorei", :species))
+```
+
+For gene functions, using a string to index will return
+all rows, regardless of the taxon.
+If you just want a single value, you can use a `GeneFunction` directly:
+
+```julia-repl
+julia> gfs_strat[GeneFunction("UniRef90_D0TRR5", "Bacteroides_dorei"), 1]
```

Comment:
Should this actually be something like this:
```
julia> gfs_strat[GeneFunction("UniRef90_D0TRR5", Taxon("Bacteroides_dorei", :species)), 1]
0.8271298594
```
When I run the line as is, I get a huge error that starts with `ERROR: BoundsError: attempt to access Axis(GeneFunction[GeneFunction("UNMAPPED", missing), `...